Add additional --sales-channel-doman-url option to "theme:dump" command

AlexBachmann opened the pull request:
For Stores with more than one storefront sales-channel installed, we want to be able to specify the theme we want to dump. Doing so using the theme ID is tedious for the developer, since it requires additional lookup and or documentation steps. In this non-breaking pull request I propose the addition of an input option that allows us to specify the theme by the sales-channel domain url instead. This pull request is an important stepping stone to make other development tasks, like `./psh.phar storefront:hot` and `./psh.phar storefront:hot-proxy` usable for development environments that work on more than one sales-channel theme at the same time.

<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?


### 2. What does this change do, exactly?


### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).


### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

mitelg commented:
hey @tinect 
this is the exact solution we implemented and which will be merged soon :wink: 
so I will close your PR. Thanks for stating out the problem in the first place :+1:

AlexBachmann committed:
Add additional --sales-channel-doman-url option to "theme:dump" command

For Stores with more than one storefront sales-channel installed, we want to be able to specify the theme we want to dump. Doing so using the theme ID is tedious for the developer, since it requires additional lookup and or documentation steps. In this non-breaking pull request I propose the addition of an input option that allows us to specify the theme by the sales-channel domain url instead. This pull request is an important stepping stone to make other development tasks, like `./psh.phar storefront:hot` and `./psh.phar storefront:hot-proxy` usable for development environments that work on more that sales-channel theme at the same time.

AlexBachmann committed:
Remove unnecessary whitespace as suggested by code sniffer

shyim commented:
Can you squash your commits?

AlexBachmann commented:
I'd love to. But I created this pull request directly in github and I cannot afford the effort to setup a real repository just to squash one commit.

bneumann97 commented:
Hey @AlexBachmann, thanks for your contribution 💙  Unfortunately, we do need a changelog file for this change and the commits need to be squashed. That why we have the checklist in the contribution template 🙂
